### PR TITLE
introduce ARM github workflows

### PR DESCRIPTION
**Pull request by aanm**
This commit adds the GH workflow to run on arm machines. This
effectively means that we can remove our travis integration and only use
GH actions from now on.

Successful run in https://github.com/cilium/cilium/actions/runs/8172084588/job/22341681445

### CODE REVIEW

File: .github/workflows/tests-runtime-arm.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+name: Unit Tests Arm
```
Comment (tklauser) [inline .github/workflows/tests-runtime-arm.yaml]: AFAICS, this workflow runs the same tests as `.github/workflows/integration-test.yaml`. Would it make sense to integrate the two and use a matrix for the `runs-on` (and make the docker installation dependent on `ubuntu-22.04-arm64`.

**Review by tklauser — COMMENTED** [review]
The `.github/workflows/integration-test.yaml` workflow needs to be adjusted to use the new paths to the `prepare.sh` and `build.sh` script.

File: .github/workflows/tests-runtime-arm.yaml (hunk 1 above)
Comment (aanm) [reply .github/workflows/tests-runtime-arm.yaml]: Yeah, I totally miss there was a file already for this. I'll adapt it.

**Review by tklauser — APPROVED** [review]
:rocket: 

Bye bye Travis :wave:

**Review by michi-covalent — APPROVED** [review]

**Pull request merged**